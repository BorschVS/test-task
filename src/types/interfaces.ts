import { ReactNode, ElementType, MouseEventHandler } from 'react';

import { SET_FORM_DATA } from '../redux/ducks/form';
import { CHEAP_VALUE, FAST_VALUE } from '../constants';

import {
  GET_FLIGHTS,
  SET_FLIGHTS,
  SET_FLIGHTS_LOADING,
  SET_FLIGHTS_ERROR,
  SET_STOPS_FILTER,
  SET_ALL_STOPS,
  SET_STOPS_FILTER_STATUS,
  SET_CATEGORY_FILTER,
} from '../redux/ducks/flights';
import { SET_CURRENT_FLIGHT, SET_IS_SHOWING } from '../redux/ducks/modal';
import {
  GET_IMAGE_FAILURE,
  GET_IMAGE_REQUEST,
  GET_IMAGE_SUCCESS,
} from '../redux/ducks/images';

// redux

export interface FlightData {
  id: string;
  searchId: string;
  price: number;
  carrier: string;
  segments: FlightSegments[];
}

export interface FlightSegments {
  origin: string;
  destination: string;
  date: string;
  stops: string[];
  duration: number;
}

export interface FlightsState {
  availableFlights: FlightData[];
  filteredFlights: FlightData[];
  stopsFilter: number[];
  stopsFilterStatus: boolean;
  flightCategoryFilter: string;
  loading: boolean;
  error: Error | string | null;
}

export interface ModalState {
  isShowing: boolean;
  flightId: string;
}

export interface FormState {
  formData: string;
}

export interface ImagesState {
  images: { [key: string]: string };
  loading: boolean;
  error: string | null;
}

export interface ReduxState {
  flights: FlightsState;
  modal: ModalState;
  form: FormState;
  images: ImagesState;
}

// /flights

// actions

export interface GetFlightsAction {
  type: typeof GET_FLIGHTS;
}

export interface SetFlightsAction {
  type: typeof SET_FLIGHTS;
  payload: FlightData[];
}

export interface SetFlightsLoadingAction {
  type: typeof SET_FLIGHTS_LOADING;
  payload: boolean;
}

export interface SetFlightsErrorAction {
  type: typeof SET_FLIGHTS_ERROR;
  payload: string;
}

export interface SetStopsFilterAction {
  type: typeof SET_STOPS_FILTER;
  payload: number[];
}

export interface SetAllStopsAction {
  type: typeof SET_ALL_STOPS;
  payload: number[];
}

export interface SetStopsFilterStatusAction {
  type: typeof SET_STOPS_FILTER_STATUS;
  payload: boolean;
}

export interface SetCategoryFilterAction {
  type: typeof SET_CATEGORY_FILTER;
  payload: typeof CHEAP_VALUE | typeof FAST_VALUE;
}

// /form

// actions

export interface SetFormDataAction {
  type: typeof SET_FORM_DATA;
  payload: string;
}

// /modal

export interface SetIsShowing {
  type: typeof SET_IS_SHOWING;
  payload: boolean;
}

export interface SetCurrentFlight {
  type: typeof SET_CURRENT_FLIGHT;
  payload: number;
}

// /images

// actions

export interface GetImageRequestAction {
  type: typeof GET_IMAGE_REQUEST;
  payload: string;
}

export interface GetImageSuccessAction {
  type: typeof GET_IMAGE_SUCCESS;
  payload: {
    imageName: string;
    imageUrl: string;
  };
}

export interface GetImageFailureAction {
  type: typeof GET_IMAGE_FAILURE;
  payload: string;
}

// Props

export interface FlightCardProps {
  flightData: FlightData;
}

export interface ModalProps {
  isShowing: boolean;
  hide: MouseEventHandler<HTMLAnchorElement>;
}

// image

export interface Image {
  imageName: string;
  alt?: string;
  component?: ElementType;
  children?: ReactNode;
  width?: number;
  height?: number;
  margin?: string;
}

// theme

export interface Theme {
  palette: {
    background: {
      default: string;
      paper: string;
    };
    text: {
      primary: string;
      secondary: string;
      disabled: string;
    };
    primary: {
      main: string;
      light: string;
      dark: string;
      contrastText: string;
    };
    secondary: {
      main: string;
      light: string;
      dark: string;
      contrastText: string;
    };
    error: {
      main: string;
    };
  };
  typography: {
    fontFamily: string;
    fontWeightMedium: number;
    fontWeightRegular: number;
    fontWeightBold: number;
  };
  breakpoints: {
    values: {
      xs: number;
      sm: number;
      md: number;
      lg: number;
      xl: number;
    };
  };
}
