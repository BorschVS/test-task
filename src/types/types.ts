import { CHEAP_VALUE, FAST_VALUE } from '../constants';

import {
  GetFlightsAction,
  SetAllStopsAction,
  SetFlightsAction,
  SetFlightsErrorAction,
  SetFlightsLoadingAction,
  SetStopsFilterAction,
  SetStopsFilterStatusAction,
  SetCategoryFilterAction,
  SetFormDataAction,
  SetIsShowing,
  SetCurrentFlight,
  GetImageRequestAction,
  GetImageSuccessAction,
  GetImageFailureAction,
} from '../types/interfaces';

// /flights

export type FlightsActions =
  | GetFlightsAction
  | SetFlightsAction
  | SetFlightsLoadingAction
  | SetFlightsErrorAction
  | SetStopsFilterAction
  | SetAllStopsAction
  | SetStopsFilterStatusAction
  | SetCategoryFilterAction;

export type FlightCategory = typeof CHEAP_VALUE | typeof FAST_VALUE;

// /form

export type FormActions = SetFormDataAction;

// /modal

export type ModalActions = SetIsShowing | SetCurrentFlight;

// /images

export type ImagesActions =
  | GetImageRequestAction
  | GetImageSuccessAction
  | GetImageFailureAction;
