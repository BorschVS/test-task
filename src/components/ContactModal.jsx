import { createPortal } from 'react-dom';
import { useSelector } from 'react-redux';

import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  IconButton,
} from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';

import { FlightCard } from 'components';
import { Form } from './Form';

const ContactModal = ({ isShowing, hide }) => {
  const availableFlights = useSelector(
    (state) => state.flights.availableFlights
  );

  const currentFlightId = useSelector((state) => state.modal.flightId);
  const currentFlightData = availableFlights.filter(
    (flight) => flight.id === currentFlightId
  );

  return isShowing
    ? createPortal(
        <Dialog
          open={isShowing}
          onClose={hide}
          aria-labelledby="form-dialog-title"
        >
          <DialogTitle id="form-dialog-title">
            <IconButton
              aria-label="close"
              onClick={hide}
              style={{ position: 'absolute', right: 8, top: 8 }}
            >
              <CloseIcon />
            </IconButton>
          </DialogTitle>
          <DialogContent sx={{marginTop: 2}}>
            <FlightCard flightData={currentFlightData[0]}/>
            <Form />
          </DialogContent>
          <DialogActions>
            <Button onClick={hide} color="primary">
              Cancel
            </Button>
          </DialogActions>
        </Dialog>,
        document.body
      )
    : null;
};

export default ContactModal;
