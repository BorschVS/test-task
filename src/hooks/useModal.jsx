import { useDispatch, useSelector } from 'react-redux';
import { setIsShowing } from '../redux/ducks/modal';

export const useModal = () => {
  const dispatch = useDispatch();
  const isShowing = useSelector((state) => state.modal.isShowing);

  const toggleModal = () => {
    dispatch(setIsShowing(!isShowing));
  };

  return {
    isShowing,
    toggleModal,
  };
};
