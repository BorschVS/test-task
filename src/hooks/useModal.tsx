import { useDispatch, useSelector } from 'react-redux';
import { setIsShowing } from '../redux/ducks/modal';
import { ReduxState } from 'interfaces';

export const useModal = () => {
  const dispatch = useDispatch();
  const isShowing = useSelector((state: ReduxState) => state.modal.isShowing);

  const toggleModal = () => {
    dispatch(setIsShowing(!isShowing));
  };

  return {
    isShowing,
    toggleModal,
  };
};
