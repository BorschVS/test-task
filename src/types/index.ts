import { CHEAP_VALUE, FAST_VALUE } from '../constants';

import {
  GetFlightsAction,
  SetAllStopsAction,
  SetFlightsAction,
  SetFlightsErrorAction,
  SetFlightsLoadingAction,
  SetStopsFilterAction,
  SetStopsFilterStatusAction,
  SetCategoryFilterAction,
  SetFormDataAction,
  SetIsShowing,
  SetCurrentFlight,
  GetImageRequestAction,
  GetImageSuccessAction,
  GetImageFailureAction,
} from 'interfaces';

// /flights

export type FlightActions =
  | GetFlightsAction
  | SetFlightsAction
  | SetFlightsLoadingAction
  | SetFlightsErrorAction
  | SetStopsFilterAction
  | SetAllStopsAction
  | SetStopsFilterStatusAction
  | SetCategoryFilterAction;

export type FlightCategory = typeof CHEAP_VALUE | typeof FAST_VALUE;

// /form

export type FormActions = SetFormDataAction;

// /modal

export type ModalActions = SetIsShowing | SetCurrentFlight;

// /images

export type ImagesActions =
  | GetImageRequestAction
  | GetImageSuccessAction
  | GetImageFailureAction;
