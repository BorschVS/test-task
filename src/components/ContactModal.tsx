import { FC } from 'react';
import { createPortal } from 'react-dom';
import { useSelector } from 'react-redux';

import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  IconButton,
} from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';

import { FlightCard, Form } from 'components';
import { ModalProps, ReduxState } from 'types/interfaces';

const ContactModal: FC<ModalProps> = ({ isShowing, hide }) => {
  const availableFlights = useSelector(
    (state: ReduxState) => state.flights.availableFlights
  );

  const currentFlightId = useSelector(
    (state: ReduxState) => state.modal.flightId
  );
  const currentFlightData = availableFlights.filter(
    (flight) => flight.id === currentFlightId
  );

  return isShowing
    ? createPortal(
        <Dialog
          open={isShowing}
          onClose={hide}
          aria-labelledby="form-dialog-title"
        >
          <DialogTitle id="form-dialog-title">
            <IconButton
              aria-label="close"
              onClose={hide}
              style={{ position: 'absolute', right: 8, top: 8 }}
            >
              <CloseIcon />
            </IconButton>
          </DialogTitle>
          <DialogContent sx={{ marginTop: 2 }}>
            <FlightCard flightData={currentFlightData[0]} />
            <Form />
          </DialogContent>
          <DialogActions>
            <Button onClick={hide} color="primary">
              Отмена
            </Button>
          </DialogActions>
        </Dialog>,
        document.body
      )
    : null;
};

export default ContactModal;
