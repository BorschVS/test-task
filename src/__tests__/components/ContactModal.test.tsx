import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import ContactModal from 'components/ContactModal';
import { useSelector } from 'react-redux';
import { useModal } from 'hooks/useModal';
import { FlightData } from 'types/interfaces';
import { useDispatch } from 'react-redux';

jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
}));

jest.mock('hooks/useModal', () => ({
    useModal: jest.fn(),
}));

jest.mock('react-dom', () => ({
    ...jest.requireActual('react-dom'),
    createPortal: (node: React.ReactNode) => node,
}));

const flightData: FlightData = {
    searchId: '11',
    id: '1',
    price: 10000,
    carrier: 'SU',
    segments: [
        {
            origin: 'City A',
            destination: 'City B',
            date: '2024-08-01T10:00:00Z',
            duration: 120,
            stops: ['SVA'],
        },
        {
            origin: 'City B',
            destination: 'City A',
            date: '2024-08-10T12:00:00Z',
            duration: 150,
            stops: ['SVA'],
        },
    ],
};

describe('ContactModal', () => {
    const mockToggleModal = jest.fn();
    const mockDispatch = jest.fn();

    beforeEach(() => {
        (useDispatch as unknown as jest.Mock).mockReturnValue(mockDispatch);
        (useSelector as unknown as jest.Mock).mockReturnValueOnce([
            flightData
        ])
            .mockReturnValueOnce('1');
        (useModal as jest.Mock).mockReturnValue({
            isShowing: true,
            toggleModal: mockToggleModal,
        });
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    test('Should render correctly when isShowing is true', () => {
        render(<ContactModal isShowing={true} hide={mockToggleModal} />);

        expect(screen.getByTestId('contact-modal')).toBeInTheDocument();
    });

    test('Should render FlightCard inside the modal', () => {
        render(<ContactModal isShowing={true} hide={mockToggleModal} />);

        expect(screen.getByTestId('flight-card')).toBeInTheDocument();
    });

    test('Should render Form inside the modal', () => {
        render(<ContactModal isShowing={true} hide={mockToggleModal} />);

        expect(screen.getByTestId('form')).toBeInTheDocument();
    });

    test('calls hide function when close button is clicked', () => {
        render(<ContactModal isShowing={true} hide={mockToggleModal} />);

        const closeButton = screen.getByLabelText('close');
        fireEvent.click(closeButton);

        expect(mockToggleModal).toHaveBeenCalledTimes(1);
    });

    test('does not render modal when isShowing is false', () => {
        render(<ContactModal isShowing={false} hide={mockToggleModal} />);

        expect(screen.queryByTestId('contact-modal')).not.toBeInTheDocument();
    });

    test('Should match FlightCard snapshot', () => {
        const { asFragment } = render(
            <ContactModal isShowing={true} hide={mockToggleModal} />);

        expect(asFragment()).toMatchSnapshot();
    });
});
