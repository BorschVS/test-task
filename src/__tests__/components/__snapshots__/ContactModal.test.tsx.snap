// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactModal Should match ContactModal snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    data-testid="contact-modal"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="form-dialog-title"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-ztsulh-MuiPaper-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
            id="form-dialog-title"
          >
            <button
              aria-label="close"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              href=""
              style="position: absolute; right: 8px; top: 8px;"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </h2>
          <div
            class="MuiDialogContent-root css-3fgj6l-MuiDialogContent-root"
          >
            <div
              class="css-1gucqoo"
              data-testid="flight-card"
            >
              <div
                class="css-1mwn02k"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1entie5-MuiTypography-root"
                >
                  10 000 Р
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1kdgtcq-MuiTypography-root"
                >
                  <span>
                    <span
                      class="MuiAvatar-root MuiAvatar-circular css-gh7mkj-MuiAvatar-root"
                    >
                      <img
                        alt="S7 Airlines company logo"
                        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                        src="/images/s7.jpg"
                      />
                    </span>
                  </span>
                  Airlines
                </p>
              </div>
              <div
                class="css-1mwn02k"
              >
                <div
                  class="css-1jibmi3"
                >
                  <p
                    class="css-9vg4wd"
                  >
                    City A - City B
                  </p>
                  <p
                    class="css-6haxrs"
                  >
                    10:00 - 12:00
                  </p>
                </div>
                <div
                  class="css-1jibmi3"
                >
                  <p
                    class="css-9vg4wd"
                  >
                    в пути
                  </p>
                  <p
                    class="css-6haxrs"
                  >
                    2Ч 0М
                  </p>
                </div>
                <div
                  class="css-1jibmi3"
                >
                  <p
                    class="css-9vg4wd"
                  >
                    1 пересадка
                  </p>
                  <p
                    class="css-6haxrs"
                  >
                    SVA
                  </p>
                </div>
              </div>
              <div
                class="css-1mwn02k"
              >
                <div
                  class="css-1jibmi3"
                >
                  <p
                    class="css-9vg4wd"
                  >
                    City B - City A
                  </p>
                  <p
                    class="css-6haxrs"
                  >
                    12:00 - 14:30
                  </p>
                </div>
                <div
                  class="css-1jibmi3"
                >
                  <p
                    class="css-9vg4wd"
                  >
                    в пути
                  </p>
                  <p
                    class="css-6haxrs"
                  >
                    2Ч 30М
                  </p>
                </div>
                <div
                  class="css-1jibmi3"
                >
                  <p
                    class="css-9vg4wd"
                  >
                    1 пересадка
                  </p>
                  <p
                    class="css-6haxrs"
                  >
                    SVA
                  </p>
                </div>
              </div>
            </div>
            <form
              class="sc-dnaUSb fXBqFE"
              data-testid="form"
            >
              <div
                class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="name"
                  id="name-label"
                >
                  Имя и фамилия
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="name"
                    name="name"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Имя и фамилия
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="phone"
                  id="phone-label"
                >
                  Телефон
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="phone"
                    name="phone"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Телефон
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="email"
                  id="email-label"
                >
                  Почта
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="email"
                    name="email"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Почта
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInfo MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInfo MuiButton-fullWidth css-tet4c5-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Отправить
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;
